src/browser/components/MainPage.jsx

```diff
@@ -20,8 +20,9 @@ import TabBar from './TabBar.jsx';
 import HoveringURL from './HoveringURL.jsx';
 import Finder from './Finder.jsx';
 import NewTeamModal from './NewTeamModal.jsx';
+import SettingsModal from './SettingsPage.jsx';
 
-export default class MainPage extends React.Component {
+export default class MainPage extends React.PureComponent {
```

Comment:
Why?
`handleBadgeChange` is called on an interval, however it also called `setState` which causes MainPage to render on the same interval.
`PureComponent` engages use of `shouldComponentUpdate` and prevents unnecessary renders.